src/librustc_mir/borrow_check/nll/region_infer/dfs.rs

```diff
@@ -34,25 +51,29 @@ impl<'tcx> RegionInferenceContext<'tcx> {
     /// - `Ok(false)` if the walk was completed with no changes;
     /// - `Err(early)` if the walk was existed early by `op`. `earlyelem` is the
     ///   value that `op` returned.
-    pub(super) fn dfs<C>(&self, mir: &Mir<'tcx>, mut op: C) -> Result<bool, C::Early>
+    #[inline(never)] // ensure dfs is identifiable in profiles
```

Comment:
I say lets leave the `#[inline(never)]` in for now, because its more important that we can get more useful profile info out-of-the-box than it is to worry about micro-optimization (the level of performance regression we are trying to address greatly dwarfs any benefit provided by inlining).

Reply:
I don't believe this would ever be criticial for performance, this is why I kept it in, but for sure it is not relevant *now*, given the data we're looking at. =) And it's convenient for profiling etc.